lib/engine/game/g_18_fr/game.rb

```diff
@@ -0,0 +1,213 @@
+
+        TRAINS = [{ name: '2', distance: 2, price: 100, rusts_on: '3P', num: 40 },
+                  { name: '2+', distance: 2, price: 100, obsolete_on: '3P', num: 4 },
+                  { name: '3+', distance: 3, price: 300, obsolete_on: 'G*D', num: 12 },
+                  { name: '3P', distance: 3, price: 400, num: 1 },
+                  { name: '2P', distance: 2, price: 300, num: 5 },
+                  { name: '5', distance: 5, price: 600, num: 6 },
+                  { name: '6*D', distance: 6, price: 800, num: 30 },
+                  { name: '2P*', distance: 2, price: 200, num: 1 }].freeze
+
+        STATUS_TEXT = Base::STATUS_TEXT.merge(
+          'two_tile_lays' => ['Two tiles', 'Corporations may lay two tiles for 20 F. One may be upgrade.'\
+                                           'Can\'t upgrade the tile just laid'],
+          'free_ports' => ['Free ports', 'Ports no longer count towards train length']
+        ).freeze
+
+        ONE_YELLOW_TILE_LAY = [{ lay: true, upgrade: false }].freeze
+        TWO_TILE_LAYS = [
+          { lay: true, upgrade: true },
+          { lay: true, upgrade: :not_if_upgraded, cost: 20, cannot_reuse_same_hex: true },
+        ].freeze
+
+        B_HEX_NAMES = %w[E8 H3].freeze
+        YELLOW_B_TILE_NAME = 'FRBY'
+        GREEN_B_TILE_NAME = 'FRBG'
+
+        def setup
+          @extra_tile_lay = false
+          @free_ports = false
+        end
+
+        def init_round
+          # skipping the initial auction for now
+          @log << "-- #{round_description('Stock', 1)} --"
+          @round_counter = 1
+          stock_round
+        end
+
+        def stock_round
+          close_bank_shorts
+          @interest_fixed = nil
+
+          G1817::Round::Stock.new(self, [
+            Engine::Step::DiscardTrain,
+            Engine::Step::HomeToken,
+            G18FR::Step::BuySellParShares,
+          ])
+        end
+
+        def operating_round(round_num)
+          @interest_fixed = interest_rate
+
+          G1817::Round::Operating.new(self, [
+            G1817::Step::Bankrupt,
+            G1817::Step::CashCrisis,
+            G1817::Step::Loan,
+            G1817::Step::SpecialTrack,
+            G1817::Step::Assign,
+            G18FR::Step::Track,
+            Engine::Step::Token,
+            Engine::Step::Route,
+            G1817::Step::Dividend,
+            Engine::Step::DiscardTrain,
+            G1817::Step::BuyTrain,
+          ], round_num: round_num)
+        end
+
+        def after_phase_change(name)
+          case name
+          when 'Green'
+            @log << '-- From now on, corporations may lay two for 20 F. Only one may be upgrade --'
+            @extra_tile_lay = true
+          when 'Blue'
+            @log << '-- From now on, corporations may lay two for 20 F. Only one may be upgrade --'
+            @free_ports = true
+          end
+        end
+
+        def tile_lays(_entity)
+          @extra_tile_lay ? TWO_TILE_LAYS : ONE_YELLOW_TILE_LAY
```

Comment:
If youŕe using the phase statuses you do not need to have a variable tracking the status of the phase changes. Instead you can query the current phase's status like this:
```suggestion
          @phase.status.include?(´two_tile_lays) ? TWO_TILE_LAYS : ONE_YELLOW_TILE_LAY
```
If you do this you will need neither the `@extra_tile_lay`/`@free_ports` variables, nor the `setup`/`after_phase_change` methods.

You can also create a methods (`two_tile_lays?`/`free_ports?`) that wrap these checks to the phase status array. That might make the code easier to read.

Reply:
Ok. If it gets used in more than one place, I'll wrap it.